test/e2e/flowaggregator_test.go

```diff
@@ -456,26 +461,92 @@ func testHelper(t *testing.T, data *TestData, podAIPs, podBIPs, podCIPs, podDIPs
 		}
 	})
 
+	// Creating test client and server Pods for ToExternal tests
+	// All three client Pods are deployed on Node 0, which is the control-plane Node
+	serverIPs, clientAName, clientAIPs, clientBName, clientBIPs, clientCName, clientCIPs := createToExternalTestPods(t, data)
```

Comment:
quick question: are we using 3 different clients to make sure we can differentiate between the records for the 3 different tests?

Reply:
If this is correct, and we don't need multiple clients for any given test, then it would really make more sense to create each client separately in its own subtest

Reply:
Yes, we use 3 clients to help differentiate records. I implemented it this way to avoid a situation where different clients are assigned the same IP when each client is created and destroyed within each subtest.

Reply:
in the unlikely event that the same IP address is reused, does it matter? The clients would have different names anyway, isn't that enough to differentiate appropriately?

Reply:
The current code lgtm. I think there is always room for improvement in the future here. For example, we could use a more generic way to match records, where the test case can specify arbitrary conditions. I recently did something similar for audit logging e2e tests: https://github.com/antrea-io/antrea/blob/918a19e7d9db5a5a2f5bae5796f331ba367073c4/test/e2e/antreapolicy_test.go#L2493. I think it would make the tests better. It could look like this:
```go
type flowRecordMatcher struct {...}
func NewFlowRecordMatcher() *FlowRecordMatcher {}
func (m *flowRecordMatcher) WithSrcPod(namespace, name string) *FlowRecordMatcher { ... }
// ...
func (m *flowRecordMatcher) MatchForFlowCollector(output []byte) bool { ... }
func (m *flowRecordMatcher) MatchForClickHouse(rows []*ClickHouseFullRow) bool { ... }
```

It would also reduce the size of the parameter list for quite a few functions, since you would only need to pass around the `*flowRecordMatcher` object.

Just an idea for a follow-up PR.

Reply:
Thanks Antonin, yes the test code looks much better in this way. I can create an issue to track this refractoring work for now.